### PR TITLE
[test] Add test for CUDA Fortran

### PR DESCRIPTION
**Pull request by ajocksch**
This check contains a workaround which will be removed in the future

### CODE REVIEW

**Comment by victorusu** [comment]
@jenkins-cscs retry all

**Comment by vkarak** [comment]
@jenkins-cscs retry all

**Review by vkarak — CHANGES_REQUESTED** [review]
Please write a stand-alone test. Don't copy paste it from another one.

File: cscs-checks/prgenv/cuda_fortran_check.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.required_version('>=2.14')
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda_fortran_check.py:10]: Remove this.

File: cscs-checks/prgenv/cuda_fortran_check.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@rfm.simple_test
+class CUDAFortranCheck(rfm.RegressionTest):
+    def __init__(self):
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'tiger:gpu']
+        self.valid_prog_environs = ['PrgEnv-pgi']
+        self.sourcepath = 'vecAdd_cuda.f90'
+
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda_fortran_check.py:17]: Remove this line.

File: cscs-checks/prgenv/cuda_fortran_check.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        self.num_tasks = 1
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda_fortran_check.py:18]: This is not needed.

File: cscs-checks/prgenv/cuda_fortran_check.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        self.modules = ['craype-accel-nvidia60']
+
+        self.executable = self.name
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda_fortran_check.py:21]: Remove this and the empty line above.

File: cscs-checks/prgenv/cuda_fortran_check.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        self.build_system = 'SingleSource'
+        self.num_gpus_per_node = 1
+        self.num_tasks_per_node = 1
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda_fortran_check.py:24]: Not needed.

File: cscs-checks/prgenv/cuda_fortran_check.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        result = sn.extractsingle(r'final result:\s+(?P<result>\d+\.?\d*)',
+                                  self.stdout, 'result', float)
+        self.sanity_patterns = sn.assert_reference(result, 1., -1e-5, 1e-5)
+
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda_fortran_check.py:28]: Remove this line.

File: cscs-checks/prgenv/cuda_fortran_check.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        self.maintainers = ['TM', 'AJ']
+        self.tags = {'production', 'craype'}
+
+    @rfm.run_before('compile')
+    def setflags(self):
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda_fortran_check.py:33]: This is the workaround so please name the hook accordingly. Check how other tests are doing it, e.g., `cscs-checks/prgenv/opencl_check.py`.

**Review by vkarak — CHANGES_REQUESTED** [review]
I'm also not sure that the location is correct, but I can take care of it.

File: cscs-checks/prgenv/cuda_fortran_check.py (hunk 7 above)
Comment (ajocksch) [reply cscs-checks/prgenv/cuda_fortran_check.py:33]: I do not understand this last comment

**Review by vkarak — APPROVED** [review]

**Pull request merged**